docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,532 @@
+
+# 1. Key Concepts
+## 1.1 Init_orca_context
+A BigDL Orca program usually starts with the initialization of OrcaContext. For every BigDL Orca program, you should call `init_orca_context` at the beginning of the program as below:
+
+```python
+from bigdl.orca import init_orca_context
+
+init_orca_context(cluster_mode, cores, memory, num_nodes, driver_cores, driver_memory, extra_python_lib, conf)
+```
+
+In `init_orca_context`, you may specify necessary runtime configurations for running the example on YARN, including:
+* `cluster_mode`: a String that specifies the underlying cluster; valid value includes `"local"`, `"yarn-client"`, `"yarn-cluster"`, `"k8s-client"`, `"k8s-cluster"`, `"bigdl-submit"`, `"spark-submit"`, etc.
+* `cores`: an Integer that specifies the number of cores for each executor (default to be `2`).
+* `memory`: a String that specifies the memory for each executor (default to be `"2g"`).
+* `num_nodes`: an Integer that specifies the number of executors (default to be `1`).
+* `driver_cores`: an Integer that specifies the number of cores for the driver node (default to be `4`).
+* `driver_memory`: a String that specifies the memory for the driver node (default to be `"1g"`).
+* `extra_python_lib`: a String that specifies the path to extra Python package, one of `.py`, `.zip` or `.egg` files (default to be `None`).
+* `conf`: a Key-Value format to append extra conf for Spark (default to be `None`).
+
+__Note__: 
+* All arguments __except__ `cluster_mode` will be ignored when using `bigdl-submit` or `spark-submit` to submit and run Orca programs.
+
+After the Orca programs finished, you should call `stop_orca_context` at the end of the program to release resources and shutdown the underlying distributed runtime engine (such as Spark or Ray).
+```python
+from bigdl.orca import stop_orca_context
+
+stop_orca_context()
+```
+
+For more details, please see [OrcaContext](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/orca-context.html).
+
+## 1.2 Yarn-Client & Yarn-Cluster
+The difference between yarn-client and yarn-cluster is where you run your Spark driver. 
+
+For yarn-client, the Spark driver runs in the client process, and the application master is only used for requesting resources from YARN, while for yarn-cluster the Spark driver runs inside an application master process which is managed by YARN on the cluster.
+
+For more details, please see [Launching Spark on YARN](https://spark.apache.org/docs/latest/running-on-yarn.html#launching-spark-on-yarn).
+
+## 1.3 Use Distributed Storage When Running on YARN
+__Note:__
+* When you are running programs on YARN, you should load data from a distributed storage (e.g. HDFS or S3) instead of the local file system.
```

Comment:
only one note, remote the bullet point and put it in one line.

should -> are recommended